dali/kernels/reduce/find_reduce_test.cu

```diff
@@ -114,22 +109,34 @@ class FindFirstLastTestGPU : public ::testing::Test {
     DynamicScratchpad dyn_scratchpad({}, AccessOrder(ctx.gpu.stream));
     ctx.scratchpad = &dyn_scratchpad;
 
-    auto out_begin = out_begin_.gpu().to_static<0>();
-    auto out_length = out_length_.gpu().to_static<0>();
+    auto out_first = out_first_.gpu().to_static<0>();
+    auto out_last = out_last_.gpu().to_static<0>();
     auto in = in_.gpu().to_static<1>();
+    int nsamples = in.size();
 
-    FindFirstLastGPU kernel;
-    kernel.template Run<T, Idx, Predicate, begin_length<Idx>>(ctx, out_begin, out_length, in,
-                                                              predicates);
+    using Predicate = threshold<float>;
+    TensorListShape<0> scalar_sh(nsamples);
+    TestTensorList<Predicate, 0> predicates;
+    predicates.reshape(scalar_sh);
+    for (int i = 0; i < nsamples; i++) {
+      *(predicates.cpu()[i].data) = Predicate(3);
+    }
+    auto predicates_gpu = predicates.gpu();
+
+    FindReduceGPU<Idx, T, Predicate, reductions::min> first;
+    FindReduceGPU<Idx, T, Predicate, reductions::max> last;
+    first.Setup(ctx, in.shape);
+    last.Setup(ctx, in.shape);
+    first.Run(ctx, out_first, in, predicates_gpu);
+    last.Run(ctx, out_last, in, predicates_gpu);
```

Comment:
So we need two kernel invocations. What is the perf difference between the old and the new approach?

Reply:
The thing is that the old approach was very inefficient for large inputs, since it used a single block per sample. The reduction kernels are able to perform multi-step reduction so we went with that approach for perf reasons. The alternative was to reinvent the wheel and replicate the multi-step reduction logic here. Let me measure the perf impact and come back to you.